### PR TITLE
feat[ft-password-reset]

### PR DESCRIPTION
**Pull request by ngoma3**
This branch will allow a user to reset his/her account password by sending a link via the user's email

## description of task

- allow user to reset password
- write its swagger documentation
- this branch contains middle ware file for setting authontications to user

## how to run it

- git clone from repo
- git checout ft-password-rese
- set .env as shown in .envexample
- run npm i
- run npm run dev
- copy this url in your postman: `http://localhost:{port}/api/v1/email`,
- chose your email and test it
- after the email verification you may receive a mail message with the token
- use the link provided in the mail inbox  which has a token as a parameter
- set your password.

feat[ft-password-reset]

### CODE REVIEW

**Review by Kevin-ishimwe — APPROVED** [review]

**Review by Evaristekanova — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# ecommerce-app-legends-bn-   ![check-code-coverage](https://img.shields.io/badge/code--coverage-86.38%25-green) [![Node.js CI](https://github.com/atlp-rwanda/ecommerce-app-legends-bn/actions/workflows/node.js.yml/badge.svg)](https://github.com/atlp-rwanda/ecommerce-app-legends-bn/actions/workflows/node.js.yml) [![Maintainability](https://api.codeclimate.com/v1/badges/92c706fad38b90146d03/maintainability)](https://codeclimate.com/github/atlp-rwanda/ecommerce-app-legends-bn/maintainability) [![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)
+<<<<<<< HEAD
```
Comment (mansurissa) [inline README.md]: Resolve conflicts in Readme.md

**Review by mansurissa — CHANGES_REQUESTED** [review]
Work on the comment asap

**Review by Aimeana100 — CHANGES_REQUESTED** [review]
great job @ngoma3 , but you will still have to handle unappropriete responses, considering the response best format and translations as well.

**Review by Aimeana100 — CHANGES_REQUESTED** [review]
great job @ngoma3 , but you will still have to handle unappropriete responses, considering the response best format and translations as well.

**Review by BenoitIT — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (BenoitIT) [reply README.md]: @ngoma3  there is infinite loading once the email is sent to the user so the user  has not known what happened after sending the request

**Review by mansurissa — APPROVED** [review]
Good now

**Pull request merged**